### PR TITLE
fix: safe access to define function params

### PR DESCRIPTION
**Pull request by hugocaillard**
Follow up for #728.
Fix more unsafe access to Slice in AST Visitor.
Plus: allow to disable/enable some LSP features

Screenshot 2022-12-19 at 16 17 44

### CODE REVIEW

File: components/clarity-vscode/package.json [hunk 1]
```diff
@@ -77,6 +77,26 @@
...
+            "description": "Allow auto completion for native and custom functions"
+          },
+          "clarity-lsp.hover": {
+            "type": "boolean",
+            "default": true,
+            "description": "Show documentation for native keywords and method on Hover"
+          },
+          "clarity-lsp.documentSymbols": {
+            "type": "boolean",
+            "default": false,
```
Comment (hugocaillard) [inline components/clarity-vscode/package.json:93]: False by default. This feature was too unstable lately compared to the benefits it brings. (Even though issues should be fixed now, it seems safer for now)

File: components/clarity-repl/src/analysis/ast_visitor.rs [hunk 2]
```diff
@@ -53,59 +53,33 @@ pub trait ASTVisitor<'a> {
...
                             match args.get(0).unwrap_or(&DEFAULT_EXPR).match_list() {
                                 Some(signature) => {
-                                    let name = signature[0].match_atom().unwrap_or(&DEFAULT_NAME);
+                                    let name = signature
+                                        .get(0)
+                                        .and_then(|n| n.match_atom())
+                                        .unwrap_or(&DEFAULT_NAME);
                                     let params = match signature.len() {
-                                        1 => None,
+                                        0 | 1 => None,
```
Comment (hugocaillard) [inline components/clarity-repl/src/analysis/ast_visitor.rs:66]: This is the actual fix

**Review by lgalabru — APPROVED** [review]
LGTM, thanks @hugocaillard!

File: components/clarity-repl/src/analysis/ast_visitor.rs [hunk 3]
```diff
@@ -53,59 +53,33 @@ pub trait ASTVisitor<'a> {
...
-                                        name,
-                                        params,
-                                        args.get(1).unwrap_or(&DEFAULT_EXPR),
-                                    );
-                                }
-                                _ => {
-                                    false;
-                                }
-                            }
-                            true
```
[the remark this answers is not in the document]
Comment (obycode) [reply components/clarity-repl/src/analysis/ast_visitor.rs:75]: That looks like a mistake. I can't think of why that would've been intentional. Thanks!

File: components/clarity-vscode/package.json (hunk 1 above)
Comment (lgalabru) [reply components/clarity-vscode/package.json:93]: is it possible / would it make sense to enable this by default only on pre-releases.
it'd be great to have some eyes on experimental features if developers are willing to use pre-release software.

**Review by obycode — APPROVED** [review]
LGTM. Thanks for this fix!

File: components/clarity-repl/src/analysis/ast_visitor.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (hugocaillard) [reply components/clarity-repl/src/analysis/ast_visitor.rs:75]: Thanks for confirming @obycode

**Pull request merged**